hoomd/BoxResizeUpdater.h

```diff
@@ -14,51 +14,84 @@
 
 #include "Updater.h"
 #include "Variant.h"
+#include "BoxDim.h"
 
 #include <memory>
+#include <string>
+#include <stdexcept>
 #include <pybind11/pybind11.h>
 
 #ifndef __BOXRESIZEUPDATER_H__
 #define __BOXRESIZEUPDATER_H__
 
-//! Updates the simulation box over time
-/*! This simple updater gets the box lengths from specified variants and sets those box sizes
-    over time. As an option, particles can be rescaled with the box lengths or left where they are.
-
-    \ingroup updaters
+/// Updates the simulation box over time
+/** This simple updater gets the box lengths from specified variants and sets
+ * those box sizes over time. As an option, particles can be rescaled with the
+ * box lengths or left where they are.
+ * \ingroup updaters
 */
 class PYBIND11_EXPORT BoxResizeUpdater : public Updater
     {
     public:
-        //! Constructor
+        /// Constructor
         BoxResizeUpdater(std::shared_ptr<SystemDefinition> sysdef,
-                         std::shared_ptr<Variant> Lx,
-                         std::shared_ptr<Variant> Ly,
-                         std::shared_ptr<Variant> Lz,
-                         std::shared_ptr<Variant> xy,
-                         std::shared_ptr<Variant> xz,
-                         std::shared_ptr<Variant> yz);
-
-        //! Destructor
+                         pybind11::object initial_box,
+                         pybind11::object final_box,
+                         std::shared_ptr<Variant> variant);
+
+        /// Destructor
         virtual ~BoxResizeUpdater();
 
-        //! Sets parameter flags
-        void setParams(bool scale_particles);
+        /// Sets particle scaling. When true, particle positions are scaled with the box.
+        void setScaleParticles(bool scale_particles)
+            {
+            m_scale_particles = scale_particles;
+            }
+
+        /// Gets particle scaling setting
+        bool getScaleParticles() {return m_scale_particles;}
+
+        /// Set a new initial box from a python object
+        void setBox1(pybind11::object initial_box)
+            {
+            m_initial_box = initial_box;
+            }
+
+        /// Get the final box
+        pybind11::object getBox1() {return m_initial_box;}
+
+        /// Set a new final box from a python object
+        void setBox2(pybind11::object final_box) {m_final_box = final_box;}
 
-        //! Take one timestep forward
+        /// Get the final box
+        pybind11::object getBox2() {return m_final_box;}
```

Comment:
I would prefer to consistently use the names `BoxInitial` and `BoxFinal` rather than `Box1` and `Box2`.